ipa-step-derive/src/sum.rs

```diff
@@ -0,0 +1,57 @@
+use std::ops::Add;
+
+use proc_macro2::{Literal, Punct, Spacing, TokenStream};
+use quote::ToTokens;
+
+/// A complex sum, comprised of multiple statements, plus a simple counter.
+/// This implements `ToTokens` so that it can be dropped into `quote!()` easily.
+#[derive(Default, Clone)]
+pub struct ExtendedSum {
```

Comment:
Where is this used?

Reply:
this is used inside the compact gate module to keep track of number of steps in a tree